ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/jca/SubordinateTxUnitTest.java

```diff
@@ -73,4 +78,85 @@ public void testAtomicAction () throws Exception
         
         assertTrue(saa2.restore_state(is, ObjectType.ANDPERSISTENT));
     }
+
+    @Test
+    public void testSAA() throws Exception {
+        SubordinateAtomicAction saa = new SubordinateAtomicAction();
+
+        saa.add(new XAResourceRecord(null, new XAResource() {
+            @Override
+            public void commit(Xid xid, boolean b) throws XAException {
+
+            }
+
+            @Override
+            public void end(Xid xid, int i) throws XAException {
+
+            }
+
+            @Override
+            public void forget(Xid xid) throws XAException {
+
+            }
+
+            @Override
+            public int getTransactionTimeout() throws XAException {
+                return 0;
+            }
+
+            @Override
+            public boolean isSameRM(XAResource xaResource) throws XAException {
+                return false;
+            }
+
+            @Override
+            public int prepare(Xid xid) throws XAException {
+                return 0;
+            }
+
+            @Override
+            public Xid[] recover(int i) throws XAException {
+                return new Xid[0];
+            }
+
+            @Override
+            public void rollback(Xid xid) throws XAException {
+
+            }
+
+            @Override
+            public boolean setTransactionTimeout(int i) throws XAException {
+                return false;
+            }
+
+            @Override
+            public void start(Xid xid, int i) throws XAException {
+
+            }
+        }, new XidImple(new Xid() {
+
+            @Override
+            public int getFormatId() {
+                return 0;
+            }
+
+            @Override
+            public byte[] getGlobalTransactionId() {
+                return new byte[0];
+            }
+
+            @Override
+            public byte[] getBranchQualifier() {
+                return new byte[0];
+            }
+        }), null));
+
+        saa.doPrepare();
+        SubordinateAtomicAction saa2 = new SubordinateAtomicAction(saa.get_uid(), true);
+        assertTrue(saa2.getXid() != null);
+
+        saa.doCommit();
+        SubordinateAtomicAction saa3 = new SubordinateAtomicAction(saa.get_uid(), true);
+        assertTrue(saa3.getXid() == null); // Since the SAA was committed the transaction log record will have been removed so the xid returned from getXid() should no longer be available and the intention is the SAA creator would disregard this instance
```

Comment:
I will still mark the PR as ready to merge but a 254 character line length is rather long!